components/function-controller/internal/webhook/defaulting_webhook.go

```diff
@@ -4,30 +4,38 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+
+	"github.com/pkg/errors"
+
 	"net/http"
 
+	serverlessv1alpha1 "github.com/kyma-project/kyma/components/function-controller/pkg/apis/serverless/v1alpha1"
+
 	serverlessv1alpha2 "github.com/kyma-project/kyma/components/function-controller/pkg/apis/serverless/v1alpha2"
 	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
 type DefaultingWebHook struct {
-	config  *serverlessv1alpha2.DefaultingConfig
-	client  ctrlclient.Client
-	decoder *admission.Decoder
+	configAlphaV2 *serverlessv1alpha2.DefaultingConfig
+	configAlphaV1 *serverlessv1alpha1.DefaultingConfig
+	client        ctrlclient.Client
+	decoder       *admission.Decoder
 }
 
-func NewDefaultingWebhook(config *serverlessv1alpha2.DefaultingConfig, client ctrlclient.Client) *DefaultingWebHook {
+func NewDefaultingWebhook(configV1Alpha1 *serverlessv1alpha1.DefaultingConfig, configV1Alpha2 *serverlessv1alpha2.DefaultingConfig, client ctrlclient.Client) *DefaultingWebHook {
 	return &DefaultingWebHook{
-		config: config,
-		client: client,
+		configAlphaV1: configV1Alpha1,
+		configAlphaV2: configV1Alpha2,
+		client:        client,
 	}
 }
+
 func (w *DefaultingWebHook) Handle(_ context.Context, req admission.Request) admission.Response {
-	if req.RequestKind.Kind == "Function" {
+	if req.Kind.Kind == "Function" {
```

Comment:
I used `Kind` because it's name of submitted object.
`RequestKind`is optional field which contains the original object before conversions. 

In this case we're interested in the final object.